lib/clients/helm.js

```diff
@@ -379,13 +388,40 @@ class Helm {
     await this._installGrafana();
     await this._installLoki();
     await this._installMatrixbot();
+  }
+
+  async _installTelemetry(index) {
     if (index === 0) {
       await this._installSubstrateTelemetry();
     }
   }
 
   async _installPrometheusOperator() {
-    return this._installGenericChart('prometheus-operator');
+    if(this.config.monitoring.opsgenie.enabled){
+      try {
+        const templatePath = "./config/charts/templates/prometheus-operator-opsgenie.tpl"
+        const templateSource = fs.readFileSync(templatePath,'utf8');
+        const template = handlebars.compile(templateSource, { strict: true })
+        const data = {
+          opsgenieEnabled: this.config.monitoring.opsgenie.enabled,
+          opsgenieToken : this.config.monitoring.opsgenie.token,
+          opsgenieUrl: this.config.monitoring.opsgenie.url
+          }
+        const prometheusConfig = template(data)
```

Comment:
We don't need all this, the `tpl` module does all of it with the `create` function https://github.com/w3f/polkadot-deployer/blob/master/lib/tpl/index.js#L13 you just need to pass the source template path, the target path and the data.